design/nfs/nfs-provisioner-controlled-by-operator.md

```diff
@@ -0,0 +1,115 @@
+# NFS Provisioner Controlled by Operator
+
+## Summary
+
+NFS Provisioner is a built in dynamic provisioner for Rook NFS. The functionality works fine but has a issue which the provisioner uses the underlying directory for each provisioned PV when provisioning two or more PV in the same share/export. This issue makes across provisioned PVs in the same share/export has the same data.
+
+This hierarchy is the current behaviour of NFS Provisioner when provisioning two PV in the same share/export.
+
+```console
+export
+├── sample-export
+|   ├── data (from PV-A)
+|   ├── data (from PV-B)
+|   ├── data (from PV-A)
+|   └── data (from PV-A)
+└── another-export
+```
+
+Both PV-A and PV-B uses the `sample-export` directory as their data location.
+
+This proposal is to make Rook NFS Provisioner to create sub-directory for each provisioned PV when provisioning two or more PV in the same share/export. So it will has a hierarchy like.
+
+```console
+export
+├── sample-export
+│   ├── pv-a
+│   │   ├── data (from PV-A)
+│   │   ├── data (from PV-A)
+│   │   └── data (from PV-A)
+│   └── pv-b
+│       └── data (from PV-B)
+└── another-export
+```
+
+Since those directories are not in the NFS Provisioner container but in NFS Server container, NFS Provisioner cannot directly create sub-directory for that. The solution is by mounting the whole of underlying NFS share/export directory so that the NFS Provisioner can create sub-directory for each provisioned PVs.
+
+### Original Issue
+
+- https://github.com/rook/rook/issues/4982
+
+### Goals
+
+- Make NFS Provisioner to create sub-directory for each provisioned PVs.
+- Make NFS Provisioner to use sub-directory for each provisioned PVs instead of use underlying directory.
+- Improve reliability of NFS Provisioner.
+
+### Non-Goals
+
+- NFS Operator manipulates uncontrolled resources.
+
+## Proposal details
+
+The approach will be similar to [Kubernetes NFS Client Provisioner](https://github.com/kubernetes-incubator/external-storage/tree/master/nfs-client) which the provisioner mounts the whole of NFS share/export into the provisioner pod (by kubelet) then the provisioner can create the appropriate sub-directory for each provisioned PVs. Currently Rook NFS Provisioner deployed independently and before NFS Server itself we cannot mount the NFS share because we don't know what NFS Server IP and the share/export directory. 
```

Comment:
```suggestion
The approach will be similar to [Kubernetes NFS Client Provisioner](https://github.com/kubernetes-incubator/external-storage/tree/master/nfs-client), where the provisioner mounts the whole of NFS share/export into the provisioner pod (by kubelet), so that the provisioner can then create the appropriate sub-directory for each provisioned PV. Currently Rook NFS Provisioner is deployed independently and before the NFS Server itself, so we cannot mount the NFS share because we don't know the NFS Server IP or the share/export directory. 
```